### PR TITLE
Main unreviewed

### PR DESCRIPTION
**Pull request by juihanamshet**

### CODE REVIEW

File: app/src/main/java/com/example/caregiver/ProfileInfo.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    /**
+     * Handle user password update.
+     * @param user The current user who is logged in the app
+     * @param newPassword User new password
+     * @param confirmPassword User new password (should be same as newPassword)
+     */
+    public void changePassword(
+            FirebaseUser user, @NonNull String newPassword, @NonNull String confirmPassword) {
+        if (!newPassword.equals(confirmPassword)) {
+            displayMessage("Your new and confirm password must be the same.", red);
```
Comment (juihanamshet) [inline app/src/main/java/com/example/caregiver/ProfileInfo.java:155]: I think this error message is usually called "Passwords do not match."

File: app/src/main/java/com/example/caregiver/HomeFragment.java [hunk 2]
```diff
@@ -36,6 +61,9 @@ public HomeFragment() {
      * @param param2 Parameter 2.
      * @return A new instance of fragment HomeFragment.
      */
+
+//    private Intent serviceIntent;
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/HomeFragment.java:65]: please remove this

File: .idea/dictionaries/juiha.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<component name="ProjectDictionaryState">
```
Comment (cassandrale179) [inline .idea/dictionaries/juiha.xml:1]: what is this dictionaries file for?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+
+    <string name="uuid_prompt">Enter UUID here</string>
+    <string name="region_prompt">Enter room name</string>
+    <string name="major_prompt">Major</string>
+    <string name="minor_prompt">Minor</string>
+    <string name="start_scanning">Start Scanning</string>
+    <string name="stop_scanning">Stop Scanning</string>
+    <string name="add_region">Add Room</string>
+
+    <!-- sign up / login -->
```
Comment (cassandrale179) [inline app/src/main/res/values/strings.xml:13]: I think if a string is only used in one component, can we stick it to the respective component only? We had merge conflict issues due to the fact that people are modifying this global file.

File: app/src/main/java/com/example/caregiver/ProfileInfo.java (hunk 1 above)
Comment (cassandrale179) [reply app/src/main/java/com/example/caregiver/ProfileInfo.java:155]: Fixed it in my latest PR.

File: app/src/main/java/com/example/caregiver/my_caregivee.java [hunk 5]
```diff
@@ -0,0 +1,268 @@
+package com.example.caregiver;
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/my_caregivee.java:1]: Can we rename this file to be CaregiveeHomePage.java to keep up with the naming convention?

File: app/src/main/java/com/example/caregiver/my_caregivee.java [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+            //Set text on text view
+            textView.setText(sChild);
+            textView.setPadding(0,10,0,10);
+            textView.setTextSize(20);
+            //Set text style Bold
+            //textView.setTypeface(null, Typeface.BOLD);
+            textView.setOnClickListener(v -> {
+                switch(i1) {
+                    case 0:
+                        ((Dashboard)getActivity()).replaceActiveFragment(new ProfileFragment());
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/my_caregivee.java:167]: You are supposed to open to the profile of the caregivee! Not the caregiver. Report bug.

File: app/src/main/java/com/example/caregiver/my_caregivee.java [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+                        break;
+                    case 1:
+                        ((Dashboard)getActivity()).replaceActiveFragment(new SetTasksFragment(caregiveeIDs.get(i)));
+                        break;
+                    case 2:
+                        Intent intent = new Intent(getContext(), ViewProgress.class);
+                        startActivity(intent);
+                        break;
+                    case 3:
+                        intent = new Intent(getContext(), RemoveCaregivee.class);
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/my_caregivee.java:177]: This class is not implemented.

File: app/src/main/java/com/example/caregiver/my_caregivee.java [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+        return v;
+    }
+
+    public class my_caregiveeAdapter extends BaseExpandableListAdapter {
+
+        private ArrayList<String> caregiveeIDs;
+        private HashMap<String, String> caregiveeIDtoNameMap;
+
+        private String[][] children = {
```
Comment (cassandrale179) [inline app/src/main/java/com/example/caregiver/my_caregivee.java:60]: Is this hardcode over here? The app crash when we saw more than 3 caregivers.

**Pull request merged**